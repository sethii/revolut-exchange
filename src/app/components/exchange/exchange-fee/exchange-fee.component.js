import React from 'react';
import { branch, renderNothing, setPropTypes, compose } from 'recompose';
import PropTypes from 'prop-types';
import getSymbolFromCurrency from 'currency-symbol-map'
import styled from 'react-emotion';

const enhance = compose(
  setPropTypes({
    fee: PropTypes.number.isRequired,
    currency: PropTypes.shape({
      code: PropTypes.string.isRequired
    }).isRequired,
    openFeeInfo: PropTypes.func.isRequired
  }),
  branch(
    ({ fee }) => fee === 0,
    renderNothing
  )
);

const FeeInfo = styled.p`
  font-size: 12px;
  color: #b1b1b1;
  font-family: Arial;
  
  & span {
    font-size: 10px;
  }
  
  & i:hover {
    cursor:pointer;
  }
`;

export const ExchangeFee = enhance(({fee, currency, openFeeInfo}) => <FeeInfo>
  Inc. fee: <span>{getSymbolFromCurrency(currency.code)}</span>{fee} <i onClick={openFeeInfo} className="fa fa-info-circle" aria-hidden="true"/>
</FeeInfo>);
