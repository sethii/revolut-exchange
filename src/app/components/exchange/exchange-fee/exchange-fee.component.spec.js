import React from 'react';
import renderer from 'react-test-renderer';
import { toClass } from 'recompose';
import { mount } from 'enzyme';

import { ExchangeFee } from './exchange-fee.component';

describe('Exchange fee component', () => {
  it('renders exchange fee', () => {
    const ExchangeFeeComponent = toClass(ExchangeFee);
    const fee = 50;
    const openFeeInfo = jest.fn();
    const currency = {
      code: 'USD'
    };

    expect(renderer.create(
      <ExchangeFeeComponent fee={fee} openFeeInfo={openFeeInfo} currency={currency}/>
    ).toJSON()).toMatchSnapshot();
  });

  it('renders null when fee is zero', () => {
    const ExchangeFeeComponent = toClass(ExchangeFee);
    const fee = 0;
    const openFeeInfo = jest.fn();
    const currency = {
      code: 'USD'
    };

    expect(renderer.create(
      <ExchangeFeeComponent fee={fee} openFeeInfo={openFeeInfo} currency={currency}/>
    ).toJSON()).toMatchSnapshot();
  });

  it('handles fee info click', () => {
    const ExchangeFeeComponent = toClass(ExchangeFee);
    const fee = 50;
    const openFeeInfo = jest.fn();
    const currency = {
      code: 'USD'
    };
    const wrapper = mount(<ExchangeFeeComponent fee={fee} openFeeInfo={openFeeInfo} currency={currency}/>);

    wrapper.find('i').simulate('click');

    expect(openFeeInfo).toBeCalled();
  });
});
