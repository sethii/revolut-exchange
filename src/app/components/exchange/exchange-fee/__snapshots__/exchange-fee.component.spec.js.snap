// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exchange fee component renders exchange fee 1`] = `
.glamor-1 {
  font-size: 12px;
  color: #b1b1b1;
  font-family: Arial;
}

.glamor-1 span {
  font-size: 10px;
}

.glamor-1 i:hover {
  cursor: pointer;
}

<p
  className="glamor-0 glamor-1"
>
  Inc. fee: 
  <span>
    $
  </span>
  50
   
  <i
    aria-hidden="true"
    className="fa fa-info-circle"
    onClick={[Function]}
  />
</p>
`;

exports[`Exchange fee component renders null when fee is zero 1`] = `null`;
